js/src/dapps/registry/ui/hash.js

```diff
@@ -14,11 +14,49 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-import React from 'react';
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import etherscanUrl from '../util/etherscan-url';
+
+import styles from './hash.css';
+
+const leading0x = /^0x/;
+
+const Hash = ({ hash, isTestnet, linked }) => {
+  let shortened = hash.toLowerCase().replace(leading0x, '');
+  shortened = shortened.length > (6 + 6)
+    ? shortened.substr(0, 6) + '...' + shortened.slice(-6)
+    : shortened;
+
+  if (linked) {
+    return (
+      <a
+        className={ styles.link }
+        href={ etherscanUrl(hash, isTestnet) }
+        target='_blank'
+      >
+        <abbr title={ hash }>{ shortened }</abbr>
+      </a>
+    );
+  }
 
-export default (hash) => {
-  const shortened = hash.length > (2 + 9 + 9)
-    ? hash.substr(2, 9) + '...' + hash.slice(-9)
-    : hash.slice(2);
   return (<abbr title={ hash }>{ shortened }</abbr>);
 };
+
+Hash.propTypes = {
+  hash: PropTypes.string.isRequired,
+  isTestnet: PropTypes.bool.isRequired,
+  linked: PropTypes.bool
+};
+
+Hash.defaultProps = {
```

Comment:
As per above. Nothing wrong with simple components, but as soon as we start doing this we are adding yet another way of writing code.